cmd/cagent/main.go

```diff
@@ -433,14 +426,32 @@ func handleFlagTest(testConfig bool, ca *cagent.Cagent) {
 			_ = sendErrorNotification("Cagent connection test failed", err.Error())
 		}
 		fmt.Printf("Cagent HUB test failed: %s\n", err.Error())
+		systemService, err := getServiceFromFlags(ca, "", "")
+		if err != nil {
+			fmt.Printf("Failed to get system service: %s\n", err.Error())
+			os.Exit(1)
+		}
+
+		status, err := systemService.Status()
+		if err != nil {
+			// service seems not installed
+			// no need to show the tip on how to restart it
+			os.Exit(1)
+		}
+
+		systemManager := service.ChosenSystem()
+		if status == service.StatusRunning || status == service.StatusStopped {
```

Comment:
Is this the check for an incorrect config?
Is `if status == service.StatusRunning` correct then? We check if the service is running, if yes we tell the user to fix the config and restart? Does the service start with an invalid config?

Reply:
> it can be running(e.g. in case hub_user/hub_password is invalid or hub_url is reachable)